engine/function/src/templates/Numeric.ftl

```diff
@@ -3378,6 +3390,600 @@ public class Numeric {
+     }
+     </#if>
+
+     <#if pt.valueType.isFloat>
+     /**
+      * Returns the floating-point number adjacent to the argument in the direction of negative infinity.
+      *
+      * @param x the value.
+      * @return the floating-point number adjacent to the argument in the direction of negative infinity.
+      *    If the value is null, returns null.
+      */
+     static public ${pt.primitive} nextDown(${pt.primitive} x) {
+        if(isNull(x)) {
+            return ${pt.null};
+        }
+
+        // skip over nulls
+        ${pt.primitive} next = Math.nextDown(x);
+        return isNull(next) ? Math.nextDown(next) : next;
+     }
+     </#if>
+
+    /**
+     * Converts an angle measured in radians to an approximately equivalent angle measured in degrees.
+     *
+     * @param x the angle in radians
+     * @return the measurement of the angle x in degrees.  If the value is null, returns null.
+     */
+    static public double toDegrees(${pt.primitive} x) {
+        if(isNull(x)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.toDegrees(x);
+    }
+
+    /**
+     * Converts an angle measured in degrees to an approximately equivalent angle measured in radians.
+     *
+     * @param x the angle in degrees
+     * @return the measurement of the angle x in radians.  If the value is null, returns null.
+     */
+    static public double toRadians(${pt.primitive} x) {
+        if(isNull(x)) {
+            return NULL_DOUBLE;
+        }
+
+        return Math.toRadians(x);
+    }
+
+    <#if pt.valueType.isInteger>
+    /**
+     * Returns the value of the argument as an int, throwing an exception if the value overflows an int.
+     *
+     * @param x the value.
+     * @return the value as an int.  If either value is null, returns null.
+     * @throws ArithmeticException if the result overflows.
+     */
+    static public int toIntExact(${pt.primitive} x) {
+        if (isNull(x)) {
+            return NULL_INT;
+        }
+
+        <#if pt.primitive == "byte" || pt.primitive == "short" || pt.primitive == "int" >
+        return x;
+        <#else>
+        return Math.toIntExact(x);
+        </#if>
+    }
+    </#if>
+
+    <#if pt.valueType.isInteger>
+    /**
+     * Returns the value of the argument as a short, throwing an exception if the value overflows a short.
+     *
+     * @param x the value.
+     * @return the value as a short.  If either value is null, returns null.
+     * @throws ArithmeticException if the result overflows.
+     */
+    static public short toShortExact(${pt.primitive} x) {
+        if (isNull(x)) {
+            return NULL_SHORT;
+        }
+
+        <#if pt.primitive == "byte" || pt.primitive == "short" >
+        return x;
+        <#else>
+        if (x > Short.MAX_VALUE || x < Short.MIN_VALUE) {
+            throw new ArithmeticException("Overflow: ${pt.primitive} value will not fit in a short" + x);
+        }
+
+        return (short) x;
```

Comment:
Similar to the above, this has a Deephaven NULL problem if the argument type is int or long and the value happens to be `Short.MIN_VALUE`

To reuse the same exception, I'd suggest writing it like

```
        short result = (short)x;
        if (x > Short.MAX_VALUE || x < Short.MIN_VALUE || isNull(result)) {
            throw new ArithmeticException("Overflow: ${pt.primitive} value will not fit in a short " + x);
        }
       return result;
```

and notice the extra space after "fit in a short"